src/content/app/tools/blast/views/blast-submission-results/BlastSubmissionResults.tsx

```diff
@@ -97,14 +102,17 @@ type SequenceBoxProps = {
 
 const SequenceBox = (props: SequenceBoxProps) => {
   const { sequence, species, blastResults } = props;
+  const parsedBlastSequence = parseBlastInput(sequence.value)[0];
+  const { header: sequenceHeader = '', value: sequenceValue } =
+    parsedBlastSequence;
+  const rulerContainer = useRef<HTMLDivElement | null>(null);
+  const { width: plotwidth } = useResizeObserver({ ref: rulerContainer });
```

Comment:
Should we consider adding a throttle to `useResizeObserver` to reduce the number of times the update gets triggered?

I noticed that the hits diagram got rendered about 300 times when I tried to resize.

Reply:
As discussed during the daily scrum, this does not look like a problem. The Resize Observer must be firing no more frequently than once per animation frame, after the rendered elements had a chance to get resized. It is not clear how the diagrams would behave if we updated them on a less frequent schedule.